routing/route/route.go

```diff
@@ -133,6 +133,11 @@ type Hop struct {
 	// LegacyPayload if true, then this signals that this node doesn't
 	// understand the new TLV payload, so we must instead use the legacy
 	// payload.
+	//
+	// NOTE: we should no longer ever create a Hop with Legacy set to true.
+	// The only reason we are keeping this member is that it could be the
+	// case that we have serialised hops persisted to disk where
+	// LegacyPayload is true.
```

Comment:
wonder when can we remove this completely🤔

Reply:
I _think_ we'd be able to do it quite soon. Afaict, we only serialise these for active payments & also for mission control. But for MC, we dont actually need this: so we should remove encoding all of this stuff for MC anyways and then for the payment scenario: as soon as no active payments have this anymore, we can remove it. So I think people just need to run this version for a bit (where no new legacy stuff gets persisted) and then after a few releases, we can just remove it.